autogpt_platform/frontend/src/app/(platform)/library/components/LibraryAgentCard/LibraryAgentCard.tsx

```diff
@@ -44,39 +44,159 @@ export default function LibraryAgentCard({
     setIsFavorite(is_favorite);
   }, [is_favorite]);
 
+  const updateQueryData = (newIsFavorite: boolean) => {
+    // Update the agent in all library agent queries
+    queryClient.setQueriesData(
+      { queryKey: ["/api/library/agents"] },
+      (oldData: any) => {
```

Comment:
To avoid untyping all the code below through `: any` casts:
```suggestion
      (oldData: InfiniteData<getV2ListLibraryAgentsResponse, number | undefined> | undefined) => {
```